Automattic/wp-calypso
# Extensions: WPJM - Update the setup status once the wizard has been visited

When the user visits any of the steps in the WPJM setup wizard, an API request is sent to set the `run_page_setup` status to `false`. `run_page_setup` is a boolean flag that determines whether or not the wizard should be shown.

Since saving the setup status is an operation that is not directly initiated by the user and is therefore completely transparent, we don’t listen for success or failure. If it succeeds, great! If not, then the user will continue to be redirected to the wizard when trying to access the settings.

## Testing

1. Follow p6r3EZ-ra-p2 to install the WPJM plugin, if applicable.
2. Log into WP Admin for your sandbox and activate the plugin.
3. `ssh` into your sandbox and `cd public_html`.
4. Enter `wp transient set _job_manager_activation_redirect 1 3600`.
5. In Calypso, go to the "Manage Plugins" page and select WP Job Manager.
6. Click on the "Edit plugin settings" link.
7. You should be redirected to the setup wizard page. For bonus points, you can also check that the following request is sent:
![wpjm-save-setup-status](https://user-images.githubusercontent.com/1190420/30986588-f3a883b4-a461-11e7-8ac4-2d2afca683b3.jpg)
8. Click the back button in your browser to return to the previous page.
9. Click on the "Edit plugin settings" link.
10. You should *not* be redirected to the setup wizard page, and instead should remain on the settings page.
